webview_panels/src/modules/documentationEditor/components/tests/EntityWithTests.tsx

```diff
@@ -66,6 +71,7 @@ const EntityWithTests = ({ title, tests, type }: Props): JSX.Element | null => {
               test={test}
               onSelect={onSelect}
               selectedTest={selectedTest}
+              className={!testKeys?.includes(test.key) ? classes.dirty : ""}
```

Comment:
if you remove and then add unique test again, it results in modified. Maybe this issue is too tricky to fix as you can have multiple unique tests as with different filters (unlikely but possible)